src/unix/special.c

```diff
@@ -37,6 +37,50 @@ static u32 null_events(file f)
     return EPOLLOUT;
 }
 
+closure_function(1, 1, void, maps_handler,
+                 buffer, b,
+                 vmap, map)
+{
+    buffer b = bound(b);
+
+    /* All mappings are assumed to be readable and private; offset, device and
+     * inode are unknown. */
+    bprintf(b, "%16lx-%16lx r%c%cp 00000000 00:00 0", map->node.r.start,
+            map->node.r.end, (map->flags & VMAP_FLAG_WRITABLE) ? 'w' : '-',
+            (map->flags & VMAP_FLAG_EXEC) ? 'x' : '-');
+
+    /* File path is unknown; only stack and heap pseudo-paths are known. */
+    if (map == current->p->stack_map) {
+        buffer_write_cstring(b, "\t[stack]");
+    } else if (map == current->p->heap_map) {
+        buffer_write_cstring(b, "\t[heap]");
+    }
+
+    buffer_write_cstring(b, "\n");
+}
+
+static sysreturn maps_read(file f, void *dest, u64 length, u64 offset)
+{
+    heap h = heap_general(get_kernel_heaps());
+    buffer b = allocate_buffer(h, 512);
+    if (b == INVALID_ADDRESS) {
+        return -ENOMEM;
+    }
+    vmap_iterator(current->p, closure(h, maps_handler, b));
```

Comment:
Looks like this closure gets leaked...you can just use a stack_closure here as a local allocation is fine (not an async operation).

Reply:
Right. Fixed by using stack_closure() as you suggest.